spot_ros2_control/examples/src/wiggle_arm.cpp

```diff
@@ -13,6 +13,14 @@
 #include "sensor_msgs/msg/joint_state.hpp"
 #include "std_msgs/msg/float64_multi_array.hpp"
 
+static const std::unordered_map<std::string, size_t> kJointNameToIndex{
+    {"front_left_hip_x", 0},  {"front_left_hip_y", 1}, {"front_left_knee", 2},   {"front_right_hip_x", 3},
+    {"front_right_hip_y", 4}, {"front_right_knee", 5}, {"rear_left_hip_x", 6},   {"rear_left_hip_y", 7},
+    {"rear_left_knee", 8},    {"rear_right_hip_x", 9}, {"rear_right_hip_y", 10}, {"rear_right_knee", 11},
+    {"arm_sh0", 12},          {"arm_sh1", 13},         {"arm_el0", 14},          {"arm_el1", 15},
+    {"arm_wr0", 16},          {"arm_wr1", 17},         {"arm_f1x", 18},
+};
+
```

Comment:
This logic should probably live in a constants file so that it can be shared with other files, it also will not work with the namespace change (the joint names there are `<namespace>/<joint_name>`.

Reply:
@mhidalgo-bdai yes, i think that's all true. Currently the joint state broadcaster that we have set up seems to publish the joint angles in a random order -- I'm not sure how to enforce that, there doesn't seem to be a param for that? It's using the `joint_state_broadcaster/JointStateBroadcaster` plugin

Reply:
If I'm not mistaken, you can tell the broadcaster which state interfaces and which joints to broadcast (see [here](https://github.com/ros-controls/ros2_controllers/blob/61f327aa43cd180fb43f792960a3d8df1f2f1e46/joint_state_broadcaster/src/joint_state_broadcaster_parameters.yaml#L7-L28)). Code is a bit convoluted but it should preserve order, I'd think.

Reply:
Hmm I just tested this out in mock mode and it didn't seem to work. I added the joints in the same order they are specified in the forward position controller config under joint state broadcaster, but this didn't preserve order. 
What I added to the config was this:
```
joint_state_broadcaster:
  ros__parameters:
    joints:
      - front_left_hip_x
      - front_left_hip_y
      - front_left_knee
      - front_right_hip_x
      - front_right_hip_y
      - front_right_knee
      - rear_left_hip_x
      - rear_left_hip_y
      - rear_left_knee
      - rear_right_hip_x
      - rear_right_hip_y
      - rear_right_knee
```
and what I was getting from joint states was this
```
$ ros2 topic echo /joint_states
header:
  stamp:
    sec: 1725049722
    nanosec: 748996370
  frame_id: ''
name:
- front_left_knee
- front_left_hip_x
- front_right_hip_y
- rear_left_hip_y
- front_right_knee
- rear_left_knee
- front_left_hip_y
- rear_right_hip_x
- front_right_hip_x
- rear_left_hip_x
- rear_right_hip_y
- rear_right_knee
...
```

Reply:
Argh, yeah, it breaks the order [here](https://github.com/ros-controls/ros2_controllers/blob/8df49cb73df207b7ac9727cdf80b30652f0ea32c/joint_state_broadcaster/src/joint_state_broadcaster.cpp#L251-L260) :roll_eyes: 

Nevermind.